app/services/fact_importer.rb

```diff
@@ -161,23 +161,25 @@ def add_new_fact(name)
   end
 
   def add_new_facts
-    facts_to_create = facts.keys - db_facts.pluck(:fact_name_id).map { |name_id| fact_names_by_id[name_id].name }
```

Comment:
Doesn't this mean we no longer need to store the `fact_names_by_id` hash?

Reply:
I wanted to be absolutely compatible with plugins which build on top of this class. But neither Ansible or Salt uses this, we shall consider removing it. Probably a Ref PR that would NOT go into 1.20 (so we merge it later on after branching)?

Reply:
These methods are all private, so they shouldn't be used in any plugins, which is why I'm fine with completely removing the hash with no deprecation.